src/ng/directive/input.js

```diff
@@ -1756,12 +1756,25 @@ var ngValueDirective = function() {
     compile: function(tpl, tplAttr) {
       if (CONSTANT_VALUE_REGEXP.test(tplAttr.ngValue)) {
         return function ngValueConstantLink(scope, elm, attr) {
-          attr.$set('value', scope.$eval(attr.ngValue));
+          var value = scope.$eval(attr.ngValue);
+          /**
+           *  input use 'value attribute' as default value if 'value property has no value'
+           *  but once set the 'value property' it will not read value from 'value attribute'
+           *  so we set both the value attribute and property here to avoid this problem.
+           */
+          attr.$set('value', value);
+          elm[0].value = value;
```

Comment:
Could you extract these two lines (along with the comment above) into a helper function to avoid duplication.